### ISSUE TITLE
NTRIP client : Is this dependent on a specific GNSS manufacturer?

### ISSUE THREAD

**Issue by joohyuk95** [ArduPilot/MAVProxy#1131]
I am using NTRIP client code [(ntrip.py)](https://github.com/ArduPilot/MAVProxy/blob/master/MAVProxy/modules/lib/ntrip.py) to use VRS-RTK.

I live in Korea. There are two NTRIP servers in Korea. 
One is the solution of the US company **Trimble** and the other is the German company **Geo++** solution.

A = **Trimble**: PIVOT
B = **Geo++** : GNSMART2

---
The issue is as follow.

If I use VRS-RTK through server B, it works without any problem.
But if I use server A, I can connect to the server, but I couldn't receive the data.
I checked if it was a server problem, but it wasn't.

To solve the problem, I first changed the NMEA_GGA message generated by the **ntrip.py** internal function 'getGGAstring' to the message generated by [NMEA generator](https://www.nmeagen.org/) and sent it to the server.
Then I could receive the data from the server A.
However, this data was not reliable. Rather, it only worsened the accuracy of standalone-gps.
This means that data is being received, but something is wrong.
I checked again if there was a problem with the server, but it was not.
And using this method, the opposite happens. (I can receive data from server A, but I cannot receive data from server B.)

The NMEA_GGA message formats are as follows.
I think the number of digits is different.
* getGGAstring func :  GPGGA,060644.00,5005.40000000,N,00839.60000000,E,1,05,0.19,+00400,M,1200.000,M,,
* NMEA generator  :  GPGGA,045338.171,3553.287,N,12836.319,E,1,12,1.0 ,0.0,M,0.0,M,,*68

In summary, if I use the ntrip.py code as it is, I can use VRS-RTK through server B, but not server A.
I don't know why this phenomenon occurs when the same protocol is used. Is the method of data parsing on the server different because the manufacturer is different? So is the ntrip.py code dependent on a specific manufacturer?

I earnestly ask for your help.

**Comment by richahert** [comment]
I have the same problem with topcon rtk corretion services. It works fine for SAPOS RTK2GO and many others.

One interesting thing I noticed is that [pygpsclient](https://github.com/semuconsulting/PyGPSClient) only worked with topcon in a newer Version because pyubx2 threw something like a `KeyError: 41 ` in the old version. Havent't understood what the problem is so far
